src/Catty/Supporting Files/AppDelegate.swift

```diff
@@ -58,22 +59,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         let vc = self.window?.rootViewController as! UINavigationController
 
         if let spvc = vc.topViewController as? ScenePresenterViewController {
-            spvc.resumeAction()
+            spvc.pauseAction()
         }
     }
 
     func applicationDidEnterBackground(_ application: UIApplication) {
-        do {
-           try AVAudioSession.sharedInstance().setActive(false)
-        } catch {
-            debugPrint("Error in AVAudioSession.sharedInstance().setActive")
-        }
+        audioEngineHelper.deactivateAudioSession()
     }
 
     func applicationWillEnterForeground(_ application: UIApplication) {
         let vc = self.window?.rootViewController as! UINavigationController
-        if let spvc = vc.topViewController as? ScenePresenterViewController {
-            spvc.resumeAction()
+        if let _ = vc.topViewController as? ScenePresenterViewController {
+            audioEngineHelper.activateAudioSession()
```

Comment:
Please revise this line. Not calling `resumeAction()` on the `ScenePresenterViewController` could potentially cause problems with the scheduler and other resources.

Reply:
Resume is already called in applicationDidBecomeActive. I was just wondering whether we need to call it twice in applicationWillEnterForeground and applicationDidBecomeActive. But you're right, maybe we need it in both methods.

Reply:
That's true. Calling it twice does not sound right. But why do you need to call  `audioEngineHelper.activateAudioSession()` separately? Is this intentional or should we call this method also in `resumeAction()`?